psydac/fem/splines.py

```diff
@@ -94,6 +94,7 @@ def __init__( self, degree, knots=None, grid=None, multiplicity=None, parent_mul
         indices = np.where(np.diff(knots[degree+1:-degree-1])>1e-15)[0]
 
         if len(indices)>0:
+            indices = indices.tolist() + [indices[-1]+1]
```

Comment:
yes, we need this when `len(indices) == 1`

Reply:
But this is done for any length >= 1... Can you explain?

Reply:
The issue when `len(indices) == 1`  is that `np.diff` function will give us an empty array, to avoid that case we append a value to it to make the length>= 2, so np.diff will give us an array of length >= 1

Reply:
I see, then you can remove your hack and simply write
```python
multiplicity = np.diff(indices).max(initial=1)
```

Reply:
Ah okay, I didn't know we have this option

Reply:
done in [c35e612](https://github.com/pyccel/psydac/pull/213/commits/c35e612c85396ad23ba720ea7b253121d32a9b71)